pkg/deploy/controller/deploymentcancellation/controller.go

```diff
@@ -0,0 +1,77 @@
+package deploymentcancellation
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// DeploymentCancellationController monitors for deployments that have been
+// cancelled by the user and sets the ActiveDeadlineSeconds to 0 on the
+// corresponding deployer pods.
+//
+// Use the DeploymentCancellationControllerFactory to create this controller.
+type DeploymentCancellationController struct {
+	// podClient provides access to pods.
+	podClient podClient
```

Comment:
I get that you only use get and update pod, but I really see no benefit in not just passing around PodsNamespacer and mocking using the fake implementation of that.  Maybe this is not the PR to do it, but my experience with this code area is that it becomes difficult to work backwards on these types of clients.